packages/edit-site/src/components/dataviews/dataviews.js

```diff
@@ -1,211 +1,54 @@
-/**
- * External dependencies
- */
-import {
-	getCoreRowModel,
-	getFilteredRowModel,
-	getSortedRowModel,
-	getPaginationRowModel,
-	useReactTable,
-} from '@tanstack/react-table';
-
 /**
  * WordPress dependencies
  */
 import {
 	__experimentalVStack as VStack,
 	__experimentalHStack as HStack,
-	VisuallyHidden,
-	DropdownMenu,
-	MenuGroup,
-	MenuItem,
 } from '@wordpress/components';
-import { useMemo } from '@wordpress/element';
-import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
-import ListView from './list-view';
-import { Pagination } from './pagination';
+import ViewList from './view-list';
+import Pagination from './pagination';
 import ViewActions from './view-actions';
 import TextFilter from './text-filter';
-import { moreVertical } from '@wordpress/icons';
-
-const EMPTY_OBJECT = {};
+import { ViewGrid } from './view-grid';
 
 export default function DataViews( {
-	actions,
-	data,
-	fields,
 	view,
 	onChangeView,
-	isLoading,
+	fields,
+	actions,
+	data,
+	isLoading = false,
 	paginationInfo,
-	options: { pageCount },
 } ) {
-	const columns = useMemo( () => {
```

Comment:
Basically all the tanstack initialization have move to the "view-list.js" file instead because we don't need it in other views like Grid, Kanban...

Reply:
Why we don't need them in the other views? Do you think all the callbacks to change the view like sorting, per page, etc.. would be different in a way?

Reply:
Well, I'm not entirely sure what you want to reuse here? by callbacks you mean things like `onChangeView( { perPage: value } )` something else? These are very small callbacks that don't need to be shared.

Reply:
I meant the callbacks like `onColumnVisibilityChange` etc.. that use `onChangeView`. But I guess the whole point here is that we define our own API with fields, etc.., use them with `setView` and that tanstack can probably be useful only in `list` view, right?

Reply:
It's not by principle or something I'm fine reusing tanstack wherever I can, I'm just not seeing where it can be useful elsewhere. Let's move forward and see. Maybe I'm missing something.